### PR TITLE
Write rotate/translate performance tests

### PR DESCRIPTION
**Pull request by OwenArnold**
These are just tests. I think these should provide a reasonable comparison against [Instrument 2.0](https://github.com/DMSC-Instrument-Data/instrument-prototype/pull/32)

### CODE REVIEW

File: Framework/Geometry/test/ComponentHelperTest.h [hunk 1]
```diff
@@ -117,4 +117,221 @@ class ComponentHelperTest : public CxxTest::TestSuite {
...
+    using namespace Mantid::Kernel;
+
+    ComponentHelper::rotateComponent(
+        *m_sansBank, *m_paramMap, m_zRotation,
+        Mantid::Geometry::ComponentHelper::Relative);
+
+    V3D pos;
+    for (int i = 1;
+         i <= static_cast<int>(m_sansInstrument->getNumberDetectors()); ++i) {
+      pos += m_sansInstrument->getDetector(i)->getPos();
```
Comment (SimonHeybrock) [inline Framework/Geometry/test/ComponentHelperTest.h]: Accessing positions this way is cheaper than the access via `MatrixWorkspace`, I think. Is this intentional?

**Comment by SimonHeybrock** [comment]
How about having benchmarks for different instrument sizes to make this more realistic? For example, 10k pixels, 100k pixels, and 1M pixels?

File: Framework/Geometry/test/ComponentHelperTest.h (hunk 1 above)
Comment (OwenArnold) [inline Framework/Geometry/test/ComponentHelperTest.h]: Yes. But it's an important point. I'll make sure that there are equivalents at the `MatixWorkspace` vs Instrument 2.0 `SpectrumInfo` level. I guess you'll be adding performance tests for the Instrument 1.0 `SpectrumInfo` too.

Comment (OwenArnold) [inline Framework/Geometry/test/ComponentHelperTest.h]: In fact I'll add a comment about that.

File: Framework/API/test/MatrixWorkspaceTest.h [hunk 2]
```diff
@@ -1486,8 +1514,51 @@ class MatrixWorkspaceTestPerformance : public CxxTest::TestSuite {
...
+    // move the bank
+    ComponentHelper::moveComponent(*m_sansBank, *m_paramMap, m_pos,
+                                   Mantid::Geometry::ComponentHelper::Relative);
+
+    V3D pos;
+    for (size_t i = 1; i < m_workspaceSans.getNumberHistograms(); ++i) {
+      pos += m_workspaceSans.getDetector(i)->getPos();
+    }
+  }
+
```
Comment (SimonHeybrock) [inline Framework/API/test/MatrixWorkspaceTest.h]: Great! I can later simply copy these tests and use `SpectrumInfo`, which should give a read speed-up, i.e., we then have three cases to compare.

File: Framework/Geometry/test/ComponentHelperTest.h [hunk 3]
```diff
@@ -117,4 +117,221 @@ class ComponentHelperTest : public CxxTest::TestSuite {
...
+
+    ComponentHelper::moveComponent(*m_sansBank, *m_paramMap, m_pos,
+                                   Mantid::Geometry::ComponentHelper::Relative);
+
+    V3D pos;
+    for (int i = 1;
+         i <= static_cast<int>(m_sansInstrument->getNumberDetectors()); ++i) {
+      pos += m_sansInstrument->getDetector(i)->getPos();
+    }
+  }
```
Comment (SimonHeybrock) [inline Framework/Geometry/test/ComponentHelperTest.h]: @OwenArnold Regarding the comparison with Google benchmark. My point was that in tests like this one we also should execute this at least a few times in a loop. To compare results we might need to do some extra math, but the tests should be more realistic then (taking account cache effects etc.).

Comment (OwenArnold) [inline Framework/Geometry/test/ComponentHelperTest.h]: @SimonHeybrock. Fair point. Shall I attempt to execute these each say 10x? I'll adjust the test names to indicate as much.

Comment (SimonHeybrock) [inline Framework/Geometry/test/ComponentHelperTest.h]: Yes, 10x sounds reasonable :+1:

**Comment by SimonHeybrock** [comment]
:shipit:

**Comment by OwenArnold** [comment]
I notice that the builds didn't get run. Probably down to Jenkins?

**Pull request merged**